go/obscuronode/enclave/sql/edgelessdb.go

```diff
@@ -0,0 +1,523 @@
+		return nil, fmt.Errorf("failed to append edb cert to mysql CA cert pool")
+	}
+	cert, err := tls.X509KeyPair([]byte(creds.UserCertPEM), []byte(creds.UserKeyPEM))
+	if err != nil {
+		return nil, fmt.Errorf("failed to prepare keypair from cert and key - %w", err)
+	}
+
+	return &tls.Config{
+		RootCAs:      caCertPool,
+		Certificates: []tls.Certificate{cert},
+		MinVersion:   tls.VersionTLS12,
+	}, nil
+}
+
+func prepareCerts() (string, string, string, error) {
+	caCert := &x509.Certificate{
+		SerialNumber:          big.NewInt(1),
+		IsCA:                  true,
+		BasicConstraintsValid: true,
+		// this subject must match the subject authorised in the manifest.json
+		Subject:   pkix.Name{CommonName: certIssuer},
+		NotBefore: time.Now(),
+		NotAfter:  time.Now().AddDate(10, 0, 0),
+		DNSNames:  []string{enclaveHostName},
+	}
+	caPrivKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		return "", "", "", fmt.Errorf("failed to generate key for CA cert to init Edgeless DB - %w", err)
+	}
+	caBytes, err := x509.CreateCertificate(rand.Reader, caCert, caCert, &caPrivKey.PublicKey, caPrivKey)
+	if err != nil {
+		return "", "", "", fmt.Errorf("failed to create CA cert - %w", err)
+	}
+	caPEMBuf := new(bytes.Buffer)
+	err = pem.Encode(caPEMBuf, &pem.Block{
+		Type:  "CERTIFICATE",
+		Bytes: caBytes,
+	})
+	if err != nil {
+		return "", "", "", fmt.Errorf("failed to create ca cert pem - %w", err)
+	}
+
+	userCert := &x509.Certificate{
+		SerialNumber: big.NewInt(2),
+		// the issuer and subject have to match those submitted in manifest.json
+		Issuer:    pkix.Name{CommonName: certIssuer},
+		Subject:   pkix.Name{CommonName: certSubject},
+		NotBefore: time.Now(),
+		NotAfter:  time.Now().AddDate(10, 0, 0),
+	}
+	certPrivKey, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		return "", "", "", fmt.Errorf("failed to generate private key for user cert - %w", err)
+	}
+	userCertBytes, err := x509.CreateCertificate(rand.Reader, userCert, caCert, &certPrivKey.PublicKey, caPrivKey)
+	if err != nil {
+		return "", "", "", fmt.Errorf("failed to prepare user certificate - %w", err)
+	}
+
+	userCertPEM := new(bytes.Buffer)
+	err = pem.Encode(userCertPEM, &pem.Block{
+		Type:  "CERTIFICATE",
+		Bytes: userCertBytes,
+	})
+	if err != nil {
+		return "", "", "", fmt.Errorf("failed to PEM encode user certificate - %w", err)
+	}
+
+	certKeyPEM := new(bytes.Buffer)
+	privKeyOut, err := x509.MarshalPKCS8PrivateKey(certPrivKey)
+	if err != nil {
+		return "", "", "", fmt.Errorf("failed to marshal cert priv key - %w", err)
+	}
+	err = pem.Encode(certKeyPEM, &pem.Block{
+		Type:  "PRIVATE KEY",
+		Bytes: privKeyOut,
+	})
+	if err != nil {
+		return "", "", "", fmt.Errorf("failed to pem encode the user private key - %w", err)
+	}
+
+	return caPEMBuf.String(), userCertPEM.String(), certKeyPEM.String(), nil
+}
+
+func prepareEDBHTTPClient(edbPEM string) (*http.Client, error) {
```

Comment:
this could to into a util.go file, and could be just a "prepareTLSHttpClient". 
It is generic HTTP over tls stuff.